projection/ibc_channel/ibc_channel.go

```diff
@@ -21,14 +23,22 @@ type IBCChannel struct {
 
 	rdbConn rdb.Conn
 	logger  applogger.Logger
+
+	config Config
+}
+
+type Config struct {
+	EnableTxMsgTrace bool
```

Comment:
Any reason we need a flag for the trace?

Reply:
I am trying to follow the way how `NFT` projection in debug mode 😂 . Maybe you could have a look on it.

The trace is not always enabled, so when using projection `IBCChannel`, it is not enabled.
Only when using projection `IBCChannelTxMsgTrace`, the trace will be enabled, which will write those msgs to the newly introduced DB table.

Reply:
Does it mean the trace is for local use only? since once the flag is off for a while, it will lose the ability to provide the full trace unless someone reindex the whole ibc_channel

Reply:
Yes, I think so. I think mainly for debugging on local and testnet. But need to confirm with Calvin. I remember he mentioned that some people might not want us to trace this on mainnet?